lib/rucio/web/rest/flaskapi/v1/locks.py

```diff
@@ -92,6 +92,45 @@ def generate(vo):
             return generate_http_error_flask(400, error)
 
 
+class LocksForManyDatasets(ErrorHandlingMethodView):
+    """ REST APIs for multiple dataset locks. """
+
+    @check_accept_header_wrapper_flask(['application/x-json-stream'])
+    def post(self):
+        """ get locks for a given scope, name.
+
+        :param scope_name: data identifier (scope)/(name).
+        :query did_type: The type used to filter, e.g., DATASET, CONTAINER.
+        :resheader Content-Type: application/x-json-stream
+        :status 200: OK.
+        :status 401: Invalid Auth Token.
+        :status 406: Not Acceptable.
+        :returns: Line separated list of dictionary with lock information.
+        """
+
+        data = json_parse(types=(dict, ))
+        try:
+            dids = data["datasets"]
+        except KeyError:
+            return 'Can not find the dataset list in the data. Use "datasets" keyword.', 400
+
+        did_type = request.args.get('did_type', default=None)
+        if did_type != 'dataset':
+            return 'Wrong did_type specified', 400
+
+        try:
+            def all_locks(dids, vo):
+                for did in dids:
+                    scope = did["scope"]
+                    name = did["name"]
+                    for lock in get_dataset_locks(scope, name, vo=vo):
```

Comment:
I think it would be better to adapt the 
- `api.locks:get_dataset_locks`
- `core.locks:get_dataset_locks`
both to be bulk queries. (And adapt the single query to just use the bulk)
Within the core method, you can still loop it like this, but this gives us at least the option later to do something smarter in the core than just looping them. (E.g. bulk query to the db, filling temporary tables and join, etc.)